### PR TITLE
Use the same golang base image to build images 

### PR DESCRIPTION
**Pull request by johscheuer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

**What this PR does / why we need it**:
Use the same golang base image to build images  and update cloudcore version in deployment manifest to latest tag. This makes the builds more consistent since they use the same golang version (if triggered at the same time and no push of a newer golang version happens). Also I updated the deployment manifest to use the latest (git) tag `1.1.0` instead of 1.0.0 which doesn't exits on docker hub.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by johscheuer** [comment]
I don't know if the change here https://github.com/kubeedge/kubeedge/pull/1271/files#diff-2bdccdce28c91bcca964f9cb02007d84R51 is worth a release note?

File: build/cloud/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+FROM golang:1.12-alpine3.9 AS builder
 
 COPY . /go/src/github.com/kubeedge/kubeedge
 
 RUN CGO_ENABLED=0 go build -v -o /usr/local/bin/cloudcore -ldflags="-w -s" \
 github.com/kubeedge/kubeedge/cloud/cmd/cloudcore
 
 
-FROM alpine:3.9
+FROM gcr.io/distroless/static
```
Comment (johscheuer) [inline build/cloud/Dockerfile]: I switched here to [distroless](https://github.com/GoogleContainerTools/distroless) which seems to be a nice fit here. My first tests where successfully. I can also move this change to another PR. I think we can also update other images to use distroless (has a smaller attack surface and requires less updates.

**Comment by johscheuer** [comment]
I reverted the change (distroless).

**Comment by johscheuer** [comment]
Oh I missed this one. The difference is that `1.12` will point to the latest `1.12` golang docker image e.g. `1.12.13` where as `1.12.1` references a fixed golang version (with all the CVE's). Also this means we should rebuilt all images after this merge since many CVE's will be resolved with building the images with a never golang version.

**Comment by fisherxu** [comment]
> Oh I missed this one. The difference is that 1.12 will point to the latest 1.12 golang docker image e.g. 1.12.13 where as 1.12.1 references a fixed golang version (with all the CVE's). Also this means we should rebuilt all images after this merge since many CVE's will be resolved with building the images with a never golang version.

Ok, make sense!
/lgtm

**Pull request merged**